### PR TITLE
Fix kokkos package.py to filter spack wrappers in launch compiler and cmake configs

### PR DESCRIPTION
**Pull request by patrickb314**
Kokkos when compiled by spack without +wrapper could potentially capture the spack compiler wrappers, resulting in cmake configs and kokkos_launch_compiler incorrectly trying to run the spack compiler wrapper after installation.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]
Waiting for maintainers to review too

**Comment by haampie** [comment]
@spackbot run pipeline

**Comment by janciesko** [comment]
LGTM

**Pull request merged**
